lib/neo4j/core/cypher_session/adaptors/bolt.rb

```diff
@@ -217,14 +217,29 @@ def flush_response
             if !(header = recvmsg(2)).empty? && (chunk_size = header.unpack('s>*')[0]) > 0
               log_message :S, :chunk_size, chunk_size
 
-              chunk = recvmsg(chunk_size)
+              chunk = flush_chunks(chunk_size)
 
               unpacker = PackStream::Unpacker.new(StringIO.new(chunk))
 
               [].tap { |r| while arg = unpacker.unpack_value!; r << arg; end }
             end
           end
 
+          def flush_chunks(first_chunk_size)
+            chunk_size = first_chunk_size
+            chunk = ''
+
+            loop do
+              chunk += recvmsg(chunk_size)
```

Comment:
It should be faster / less memory hungry to do `chunk.concat(recvmsg(chunk_size))` because it will change the string in-place rather than making a copy and re-assigning it.